src/swap_utils.c

```diff
@@ -38,6 +43,11 @@ bool parse_swap_config(const uint8_t *config, uint8_t config_len, char *ticker,
     if (config_len - offset < 1) {
         return false;
     }
-    *decimals = config[offset];
+    *decimals = config[offset++];
```

Comment:
Do in 2 lines, this kind of increment inside operations is always a pain to debug / see errors